### ISSUE TITLE
cifar100 HP with RandAug

### ISSUE THREAD

**Issue by Justin900429** [SHI-Labs/Compact-Transformers#23]
Hi, can you share the HP that your team trained it on RandAug with Timm? I have followed the same steps but still stuck on 76.9% and couldn't go any further (trained for 300 epochs should reach approximately 80%). Thanks for your help.

**Comment by alihassanijr** [comment]
Hi. Thank you for your interest.
We use `rand-m9-mstd0.5-inc1` as the augmentation policy.
There's also `--crop-pct` which for ImageNet is 0.875 (224x224 center crop out of 256x256 for eval), which is not common practice for CIFAR-10/100, specifically because of the low resolution. Therefore, set `--crop-pct` to `1.0`.
There's also a crop scale (`--scale`) for training which by default is `[0.08, 1.0]`, we set it to `[0.8, 1.0]`.
Make sure to apply CIFAR-100's mean and std as well, otherwise it will default to ImageNet's:
`--mean 0.5071 0.4867 0.4408`
`--std 0.2675 0.2565 0.2761`

With these arguments, CCT-7/3x1 with the default LR (`5e-4`), weight decay (`5e-2`) and batch size of 128, will break 80% top-1 easily after 300 epochs (default warmup & cooldown).

(Needless to say, `--img-size 32`.)

Let me know if you have any more questions.

**Comment by Justin900429** [comment]
Hi, thanks for the reply. Do you directly use the `train.py` provided by timm or just adapted the code from them?

**Comment by alihassanijr** [comment]
Hi.
We only added CIFAR-10/100 datasets to the script, along with some other minor modifications (just for our testing purposes).

**Comment by Justin900429** [comment]
I see. I wonder what optimizer you use when you trained the model? Because the default setting in timm is `SGD`. However the setting in this repo is `AdamW`. Another would be the parameters regarding to the `cutmix + mixup`, could you please provide the HP for the setting? Thank you!

**Comment by alihassanijr** [comment]
Yes, thank you for pointing that out. We did in fact use AdamW. We did not change the arguments related to augmentation.

**Comment by Justin900429** [comment]
Ok, thank you. I will take it a try. I'll close this issue and reopen it if I have another problem.
